crates/typst-cli/src/update.rs

```diff
@@ -0,0 +1,359 @@
+use std::{
+    env,
+    ffi::OsStr,
+    fs,
+    io::{BufReader, Cursor, ErrorKind, Read},
+    path::Path,
+};
+
+use semver::Version;
+use serde::Deserialize;
+use typst::{diag::bail, diag::StrResult, eval::eco_format};
+
+use crate::args::UpdateCommand;
+
+// these might not be very usefull but does make it easier to maintain
+// if the organization/repo moves or changes (only used in release fetching)
+const TYPST_GITHUB_ORG: &str = "typst";
+const TYPST_REPO: &str = "typst";
+
+#[cfg(feature = "no-self-update")]
+pub(crate) const NEVER_SELF_UPDATE: bool = true;
```

Comment:
At least at first blush I don't think that this is quite the best approach to disabling the feature. I'm guessing this is modeled after `rustup`. The entire point of that tool is to run updates, so it makes sense to still build all the functionality but just disable some features using a boolean flag. I don't think that makes sense for this situation: instead none of the update functions should even be compiled into the binary at all and the relevant bits of the man page and help output generated should also just not expose such a feature. As such I don't think you need a constant, just use configure flags to exclude al the relevant bits from even getting built.

Reply:
Although I understand why it should not be compiled I'm not going to include that just yet. This PR is still a rough concept and I don't want to spend too much time on it until I have a solid go-ahead from a maintainer. If I get that go-ahead I'll actually make sure all of the update bits won't get compiled when the feature is disabled.

Reply:
I wonder whether we should instead keep _the command,_ but not its implementation. That way, the `update` command can print a helpful message so that the user knows the binary isn't self-update-capable. Maybe it's not necessary, idk.

Reply:
I like that idea, I'll try and include that.